WooCommerce/WooCommerceUITests/Tests/LoginTests.swift

```diff
@@ -31,16 +31,13 @@ class LoginTests: XCTestCase {
         XCTAssert(WelcomeScreen.isLoaded())
     }
 
-    func testLoginOptionsArePresent() {
-        //verify tapping "Log in" will show options to enter a WordPress.com email, log in with Google, or enter your site address.
-        WelcomeScreen()
+    func testEnterEmailAndSendLinkOrEnterPassword() {
+        //enter email and click next to verify presence of send link and password option
+        _ = WelcomeScreen()
         .selectLogin()
-
-        //verify all login options exist
-        .doAllLoginOptionsExist()
-        .goBackToWelcomeScreen()
+        .proceedWith(email: TestCredentials.emailAddress)
+        .goBack()
 
         XCTAssert(WelcomeScreen.isLoaded())
```

Comment:
Should I keep coming back to the Welcome screen like in the first test or should there be a teardown method that brings me back to the welcome screen after each login test? I would vote for the second option.

Reply:
To my thinking, _if_ the app restarts b/t each test, there is no reason to go back to the welcome screen more than once. If there are multiple tests in which we need to visit this screen and go back, then I would agree a teardown method might be appropriate.

Reply:
Ok thank you, I will work on that next then. I don't like the repeated `goBack` steps I wrote to clean up after, it looks sloppy to me and it's slower.